File: src/services/coproperty-management/Myb.Coproperty/GraphQL/Mutations/AssemblyMutations.cs
using HotChocolate;
using HotChocolate.Types;
using Myb.Coproperty.Models;
using Myb.Coproperty.Services;
using Myb.Coproperty.GraphQL.Types;
using AssemblyStatusModel = Myb.Coproperty.Models.AssemblyStatus;

namespace Myb.Coproperty.GraphQL.Mutations;

[ExtendObjectType("Mutation")]
public class AssemblyMutations
{
    public async Task<Assembly> CreateAssembly(
        CreateAssemblyInput input,
        [Service] IAssemblyService assemblyService)
    {
        var assembly = input.ToEntity();
        return await assemblyService.CreateAsync(assembly);
    }

    public async Task<Assembly> UpdateAssembly(
        Guid id,
        UpdateAssemblyInput input,
        [Service] IAssemblyService assemblyService)
    {
        var assembly = await assemblyService.GetByIdAsync(id);
        if (assembly == null)
            throw new InvalidOperationException($"Assembly with ID {id} not found");

        if (input.Title != null) assembly.Title = input.Title;
        if (input.MeetingDate.HasValue) assembly.MeetingDate = input.MeetingDate.Value;
        if (input.Location != null) assembly.Location = input.Location;
        if (input.Agenda != null) assembly.Agenda = input.Agenda;
        if (input.Minutes != null) assembly.Minutes = input.Minutes;
        if (input.AssemblyType.HasValue) assembly.AssemblyType = input.AssemblyType.Value;
        if (input.Status.HasValue) assembly.Status = input.Status.Value;
        
        assembly.UpdatedAt = DateTime.UtcNow;
        await assemblyService.UpdateAsync(assembly);
        return assembly;
    }

    public async Task<bool> DeleteAssembly(
        Guid id,
        [Service] IAssemblyService assemblyService)
    {
        await assemblyService.DeleteAsync(id);
        return true;
    }

    public async Task<Assembly> UpdateAssemblyStatus(
        Guid id,
        AssemblyStatusModel status,
        [Service] IAssemblyService assemblyService)
    {
        var assembly = await assemblyService.GetByIdAsync(id);
        if (assembly == null)
            throw new InvalidOperationException($"Assembly with ID {id} not found");

        assembly.Status = status;
        assembly.UpdatedAt = DateTime.UtcNow;
        await assemblyService.UpdateAsync(assembly);
        return assembly;
    }
}

File: src/front/myb.front/libs/coproperty-module/src/lib/graphql/mutations/assembly.mutations.ts
import { gql } from 'apollo-angular';

export const CREATE_ASSEMBLY = gql`
  mutation CreateAssembly($assembly: AssemblyInput!) {
    createAssembly(assembly: $assembly) {
      id
      copropertyId
      title
      meetingDate
      location
      agenda
      assemblyType
      status
      createdAt
    }
  }
`;

export const UPDATE_ASSEMBLY = gql`
  mutation UpdateAssembly($id: UUID!, $assembly: AssemblyInput!) {
    updateAssembly(id: $id, assembly: $assembly) {
      id
      copropertyId
      title
      meetingDate
      location
      agenda
      minutes
      assemblyType
      status
      updatedAt
    }
  }
`;

export const UPDATE_ASSEMBLY_STATUS = gql`
  mutation UpdateAssemblyStatus($id: UUID!, $status: AssemblyStatus!) {
    updateAssemblyStatus(id: $id, status: $status) {
      id
      status
      updatedAt
    }
  }
`;

export const DELETE_ASSEMBLY = gql`
  mutation DeleteAssembly($id: UUID!) {
    deleteAssembly(id: $id)
  }
`;
